### DISCUSSION TITLE
Avoid showing confirmation when game is closed by the user

### DISCUSSION THREAD

**Discussion by Buzzboss** [ppy/osu#25826] [Ideas] [answered]
haveing to confirm to close the game is kinda annoying just extra steps i would like it if the confirmation dosnt always appear like when doing alt+f4 and when restarting after an update. not required but would be nice just makes things faster :)

**Comment by peppy** [comment]
It should already only be shown when it really needs to. What's the case you're hitting?

**Reply by Buzzboss** [reply]
just closeing the game and restarting after an update having to confirm to close the game just gets annoying

**Reply by Skubers** [reply]
I believe that since you deliberately click on the "Update is downloaded and ready to install" button, you confirm that yes, I want to install the update - there's no need for the quit dialog.

I remember this being tackled like 2 years ago and something was said that the quit dialog is "binded" to the quitting action so it always shows when game is trying to quit? I might be wrong though

**Reply by Buzzboss** [reply]
yeah things that you already mean to confirm (updating and installing and simply just clicking alt +4), it still give you the extra "are you sure you want to exit osu!?" confirmation

**Reply by marioman63** [reply]
one situation where it would be nice to not have to confirm to close (or at least not tab back into the game to do so) is if i click close from the task bar. In stable, I could click close on the taskbar twice (once to bring up the dialogue menu in game, again to shut the game down). In lazer, clicking once brings up the in game menu, but since you now have to hold the quit button to quit, it is impossible to close via taskbar while the game is minimized even if you right click the task bar icon again and select close.

**Reply by peppy** [reply]
> but since you now have to hold the quit button to quit

Can you show a video of this? It should like it's asking for confirmation **because you're about to lose data**. I can't reproduce this, and can close the game by hitting exit twice.

> just closeing the game and restarting after an update having to confirm to close the game just gets annoying

The game should not prompt for confirmation when installing an update. Can you provide a video of this?

**Reply by Skubers** [reply]
About the prompt during an update, there you go @peppy 


https://github.com/ppy/osu/assets/82095491/de033ea0-cb21-4ca6-ad43-a4427aba0792

**Reply by Buzzboss** [reply]
the text says click to restart, so it sould restart without have to confirm it again

**Reply by peppy** [reply, marked as answer]
Attempting to resolve this via #26558.
